WooCommerce/Classes/POS/Presentation/PointOfSalePaymentsTestView.swift

```diff
@@ -0,0 +1,52 @@
+import SwiftUI
+import Combine
+
+// Required for making a dummy order
+import Yosemite
+import Fakes
+
+class PointOfSalePaymentsTestViewModel: ObservableObject {
+    let cardPresentPayments: CardPresentPayments
+    var cancellables = Set<AnyCancellable>()
+
+    @Published var onboardingViewModels: InPersonPaymentsViewModel?
+
+    @Published var paymentModalViewModel:
+    WrappedCardPresentPaymentsModalViewModel?
+
+    init(siteID: Int64) {
+        self.cardPresentPayments = CardPresentPaymentsAdaptor(siteID: siteID)
+        cardPresentPayments.paymentScreenEventPublisher
+            .print("🅿️ payment event")
+            .map { event -> InPersonPaymentsViewModel? in
```

Comment:
Are we creating a new instance of `InPersonPaymentsViewModel` every time that this event happens and is passed as an associated type? Would it be any concern if that's the case? I don't see anything that could cause troubles but I'm wondering: If it's a new instance we have to be careful when observing changes, or losing the reference may imply stop listening to changes.

Reply:
We are, and I'm not 100% sure if it's a concern yet... I don't think it is though. 

It's poorly named – `InPersonPaymentsViewModel` is essentially `OnboardingScreenViewModel` – so it should be pretty simple, short lived, and safe to make new ones when the onboarding screen changes.